core/src/main/java/com/datastax/oss/driver/api/core/session/throttling/RequestThrottler.java

```diff
@@ -56,4 +56,10 @@ public interface RequestThrottler extends Closeable {
    * perform time-based eviction on pending requests.
    */
   void signalTimeout(@NonNull Throttled request);
+
+  /**
+   * Signals that a request has been cancelled. This indicates to the throttler that another request
```

Comment:
It took me a little bit to come around to why we couldn't just reuse an existing api method (much like the originator of the JIRA re-used `signalTimeout`), but I think i agree that it makes sense that we add a new API method.

Calling `future.cancel()` does not actually cancel an inflight query, which I suppose is a similar story to `signalTimeout` in that C* nodes may still not be working on that query.

I think it's useful to make this a separate method so throttling implementations can make their own decision about how to handle cancel. :+1: